utils/build_swift/tests/expected_options.py

```diff
@@ -0,0 +1,494 @@
+    'skip_test_watchos': True,
+    'skip_test_watchos_host': True,
+    'skip_test_watchos_simulator': True,
+    'stdlib_deployment_targets': [
+        'macosx-x86_64',
+        'iphonesimulator-i386',
+        'iphonesimulator-x86_64',
+        'appletvsimulator-x86_64',
+        'watchsimulator-i386',
+        'iphoneos-armv7',
+        'iphoneos-armv7s',
+        'iphoneos-arm64',
+        'appletvos-arm64',
+        'watchos-armv7k'
+    ],
+    'swift_analyze_code_coverage': 'false',
+    'swift_assertions': True,
+    'swift_build_variant': 'Debug',
+    'swift_compiler_version': None,
+    'swift_stdlib_assertions': True,
+    'swift_stdlib_build_variant': 'Debug',
+    'swift_tools_max_parallel_lto_link_jobs': 0,
+    'swift_user_visible_version': '4.1',
+    'symbols_package': None,
+    'test': None,
+    'test_optimize_for_size': None,
+    'test_optimized': None,
+    'tvos': False,
+    'tvos_all': False,
+    'validation_test': None,
+    'verbose_build': False,
+    'watchos': False,
+    'watchos_all': False
+}
+
+
+# -----------------------------------------------------------------------------
+
+class _BaseOption(object):
+
+    def __init__(self, option_string, dest, default=None):
+        self.option_string = option_string
+        self.dest = dest
+
+        if default is None:
+            default = EXPECTED_DEFAULTS.get(dest, None)
+
+        self.default = default
+
+    def sanitized_str(self):
+        if self.option_string.startswith('--'):
+            return self.option_string[2:].replace('-', '_')
+
+        if len(self.option_string) == 2 and self.option_string[0] == '-':
+            return self.option_string[1]
+
+        raise ValueError('invalid option_string format: ' + self.option_string)
+
+
+class Option(_BaseOption):
+    """Option that accepts no arguments."""
+
+    def __init__(self, *args, **kwargs):
+        self.value = kwargs.pop('value', None)
+        super(Option, self).__init__(*args, **kwargs)
+
+
+class ToggleOption(_BaseOption):
+    """Option that accepts no argument or an optional bool argument."""
+
+    pass
+
+
+class ChoicesOption(_BaseOption):
+    """Option that accepts an argument from a predifined list of choices."""
+
+    def __init__(self, *args, **kwargs):
+        self.choices = kwargs.pop('choices', None)
+        super(ChoicesOption, self).__init__(*args, **kwargs)
+
+
+class IntOption(_BaseOption):
+    """Option that accepts an int argument."""
+
+    pass
+
+
+class StrOption(_BaseOption):
+    """Option that accepts a str argument."""
+
+    pass
+
+
+class PathOption(_BaseOption):
+    """Option that accepts a path argument."""
+
+    pass
+
+
+class AppendOption(_BaseOption):
+    """Option that can be called more than once to append argument to internal
+    list.
+    """
+
+    pass
+
+
+class UnsupportedOption(_BaseOption):
+    """Option that is not supported."""
+
+    def __init__(self, *args, **kwargs):
+        kwargs['dest'] = kwargs.get('dest')
```

Comment:
What are you trying to do here?

Reply:
That line facilitates allowing instances of `UnsupportedOption` to skip passing in the `dest` argument since it doesn't seem necessary for that sort of option. I would be better written as `kwards['dest'] = kwargs.pop('dest', None)` which pops `dest` off the input keyword args or resolves to None and sets the value manually when passed into the superclass constructor.